include/libcyphal/transport/can/svc_rx_sessions.hpp

```diff
@@ -0,0 +1,169 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_TRANSPORT_CAN_SVC_RX_SESSIONS_HPP_INCLUDED
+#define LIBCYPHAL_TRANSPORT_CAN_SVC_RX_SESSIONS_HPP_INCLUDED
+
+#include "delegate.hpp"
+#include "libcyphal/transport/svc_sessions.hpp"
+
+#include <canard.h>
+
+#include <chrono>
+
+namespace libcyphal
+{
+namespace transport
+{
+namespace can
+{
+
+/// Internal implementation details of the CAN transport.
+/// Not supposed to be used directly by the users of the library.
+///
+namespace detail
+{
+
+/// @brief A template class to represent a service request/response RX session (both for server and client sides).
+///
+/// @tparam Interface_ Type of the session interface.
+///                    Could be either `IRequestRxSession` or `IResponseRxSession`.
+/// @tparam Params Type of the session parameters.
+///                Could be either `RequestRxParams` or `ResponseRxParams`.
+/// @tparam TransferKind Kind of the service transfer.
+///                      Could be either `CanardTransferKindRequest` or `CanardTransferKindResponse`.
+///
+template <typename Interface_, typename Params, CanardTransferKind TransferKind>
+class SvcRxSession final : public Interface_, private IRxSessionDelegate
```

Comment:
Both service RX sessions (rx request and rx response) are almost the same in terms of structure and functionality. The only slight difference is their interface name, params type and "kind" of canard transfer. So, IMO making template here is fine and justifiable b/c ...:
1. it's `detail`-s, not exposed to a user
2. no code duplication
3. both instantiations (see below just after the template) are 100% covered by unit tests (at line, *function* and *branches* level).

Reply:
BTW, `MessageRxSession` is also very similar, BUT still has its differences (in terms of params validation and anonymous case handling), so it does not fit into such RX template IMO (hence there is minor code duplication - all covered of course by tests).